src/Microsoft.AspNetCore.Mvc.RazorPages/Internal/PageActionDescriptorChangeProvider.cs

```diff
@@ -39,51 +42,69 @@ public class PageActionDescriptorChangeProvider : IActionDescriptorChangeProvide
                 throw new ArgumentNullException(nameof(razorPagesOptions));
             }
 
-            _fileProvider = fileProviderAccessor.FileProvider;
+            _watchForChanges = razorViewEngineOptions.Value.AllowUpdatingViewsOnChange;
+            if (_watchForChanges)
+            {
+                (_fileProvider, _searchPatterns, _additionalFilesToTrack) = Initialize();
+            }
 
-            var rootDirectory = razorPagesOptions.Value.RootDirectory;
-            Debug.Assert(!string.IsNullOrEmpty(rootDirectory));
-            rootDirectory = rootDirectory.TrimEnd('/');
+            (IFileProvider, string[] searchPatterns, string[] additionalFilesToTrack) Initialize()
```

Comment:
Suggest being consistent about naming the `ValueTuple`'s fields. Either name them all (which seems to be recommended practice -- at least for public methods, or none of them. As-is this tuple has fields `Item1, `searchPatterns` and `additionalFilesToTrack`.